starlette/applications.py

```diff
@@ -115,13 +115,14 @@ def debug(self) -> bool:
     @debug.setter
     def debug(self, value: bool) -> None:
         self._debug = value
-        self.middleware_stack = self.build_middleware_stack()
 
     def url_path_for(self, name: str, **path_params: typing.Any) -> URLPath:
         return self.router.url_path_for(name, **path_params)
 
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         scope["app"] = self
+        if self.middleware_stack is None:
+            self.middleware_stack = self.build_middleware_stack()
```

Comment:
Note that this will run when the lifespan is triggered most of the time but if the lifespan is disable it will run on the first request.

Reply:
I've mentioned this in the description as well. :+1: